#
# For PPC64LE only!
#
# This is fixed upstream by the removal of Versions.def
# and auto-generation of the SHLIB_COMPAT required entries.
# See: https://sourceware.org/ml/libc-alpha/2014-02/msg00818.html
# Backporting that infrastructure to RHEL 7.x is too much work
# at this junction for little reward. Instead we simply fix up
# the Versions.def to include GLIBC_2.3 which is used by
# nptl/old_pthread_atfork.c, otherwise ppc64le will get
# pthread_atfork in libpthread.so.0 when it should not.
#
# The ABI testing for libpthread.so now passes for ppc64le.
#
diff -urN glibc-2.17-c758a686/Versions.def glibc-2.17-c758a686/Versions.def
--- glibc-2.17-c758a686/Versions.def	2014-06-02 21:13:12.000000000 +0000
+++ glibc-2.17-c758a686/Versions.def	2014-06-02 21:14:38.000000000 +0000
@@ -92,6 +92,7 @@
   GLIBC_2.2
   GLIBC_2.2.3
   GLIBC_2.2.6
+  GLIBC_2.3
   GLIBC_2.3.2
   GLIBC_2.3.3
   GLIBC_2.3.4
@@ -99,6 +100,7 @@
   GLIBC_2.6
   GLIBC_2.11
   GLIBC_2.12
+  GLIBC_2.17
   GLIBC_PRIVATE
 }
 libresolv {
